Lesson19/aysel/aysel.js

```diff
@@ -0,0 +1,290 @@
+const sentenceEx5 = 'Hello, how are you?';
+const charToRemove = 'o';
+
+function removeOccurrences(string, charToRemove) {
+    if (string === '') {
+        return '';
+    }
+
+    if (string[0] === charToRemove) {
+        return removeOccurrences(string.slice(1), charToRemove);
+    } else {
+        return string[0] + removeOccurrences(string.slice(1), charToRemove);
+    }
+}
+
+console.log(removeOccurrences(sentenceEx5, charToRemove));
+
+// 5. Write a recursive function to check if an array includes a specific value.
+const numbers = [1, 2, 3, 4, 5];
+const valueToCheck = 3;
+function findSpecificValue(array, valueToCheck) {
+    if (array.length === 0) {
+        return false;
+    }
+
+    if (array[0] === valueToCheck) {
+        return true;
+    }
+
+    return findSpecificValue(array.slice(1), valueToCheck);
+}
+
+console.log(findSpecificValue(numbers, valueToCheck));
+
+
+// 6. Write a recursive function to flatten an object with nested objects into a single-level object.
+const nestedObj = {
+	name: 'John',
+	age: 30,
+	address: {
+		street: '123 Main St',
+		city: 'New York',
+		country: 'USA',
+	},
+	hobbies: ['reading', 'coding'],
+	social: {
+		twitter: '@johndoe',
+		linkedIn: 'johndoe',
+	},
+};
+
+function flattenObject (obj, parentKey = '', result = {}) {
+	for (let key in obj) {
+	  if (obj.hasOwnProperty(key)) {
+		const newKey = parentKey ? `${parentKey}.${key}` : key;
+		if (typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+		  flattenObject(obj[key], newKey, result);
+		} else {
+		  result[newKey] = obj[key];
+		}
+	  }
+	}
+	return result;
+  };
+  console.log(flattenObject(nestedObj));
+  
+
+// 7. Write a recursive function to find the maximum depth of a nested object.
+const nestedObjEx7 = {
+	a: 1,
+	b: {
+		c: 2,
+		d: {
+			e: 3,
+			f: {
+				g: 4,
+			},
+		},
+	},
+	h: 5,
+};
+const maxDepth = (obj) => {
+	let max = 1; 
+	for (let key in obj) {
+	  if (typeof obj[key] === 'object' && obj[key] !== null) {
+		const depth = maxDepth(obj[key]) + 1;
+		max = Math.max(max, depth);
+	  }
+	}
+
+	return max;
+  };
+  
+
+  console.log(maxDepth(nestedObjEx7));
+  
+// 8. Write a recursive function to reverse the order of words in a sentence.
+const sentenceEx8 = 'Hello, how are you?';
+const reverseWords = (sentence) => {
+	const words = sentence.split(' ');
+  
+	const reversedWords = [];
+  
+	for (let i = words.length - 1; i >= 0; i--) {
+	  reversedWords.push(words[i]);
+	}
+  
+	return reversedWords.join(' ');
+  };
+  
+  const sentenceEx8 = 'Hello, how are you?';
+
+  const reversedSentence = reverseWords(sentenceEx8);
+  console.log(reversedSentence);
+  
+
+
+// 9. Write a recursive function to find the length of the longest word in a sentence.
+const sentenceEx9 = 'The quick brown fox jumps over the lazy dog';
+
+const findLongestWordLength = (sentence) => {
+
+	const words = sentence.split(' ');
+  
+	let maxLength = 0;
+  
+
+	for (let i = 0; i < words.length; i++) {
```

Comment:
Aysel, can you try solving this exercises using recursion instead of loop? Don't forget to pass the modified array (array without the words that were already checked) to the next round.